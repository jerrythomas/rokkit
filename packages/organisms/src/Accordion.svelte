<script>
	import { createEmitter } from '@rokkit/core'
	// import { createEventDispatcher } from 'svelte'
	import { defaultFields } from '@rokkit/core'
	import { navigator } from '@rokkit/actions'
	import { Item, Summary } from '@rokkit/molecules'
	// import List from './List.svelte'
	import ListItems from './ListItems.svelte'

	// const dispatch = createEventDispatcher()

	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 * @property {any} [items]
	 * @property {any} [fields]
	 * @property {any} [using]
	 * @property {boolean} [autoClose]
	 * @property {any} [value]
	 */

	/** @type {Props} */
	let {
		class: className = '',
		items = $bindable([]),
		fields = $bindable({}),
		using = $bindable({}),
		autoClose = false,
		value = $bindable(null),
		...events
	} = $props()
	let cursor = $state([])

	let emitter = $derived(createEmitter(events, ['collapse', 'change', 'expand', 'click']))
	let internalFields = $derived({ ...defaultFields, ...fields })
	let internalUsing = $derived({ default: Item, ...using })

	function handle(event) {
		value = event.detail.node
		cursor = event.detail.path
		if (['collapse', 'expand'].includes(event.type)) {
			if (autoClose) {
				items.map((x) => {
					if (x !== value && x[internalFields.isOpen]) {
						x[internalFields.isOpen] = false
					}
				})
			}
			items = items
		}
		emitter[event.type](event.detail)
		// dispatch(event.type, event.detail)
	}
</script>

<!-- svelte-ignore a11y_no_noninteractive_tabindex -->
<accordion
	class="flex w-full select-none flex-col {className}"
	tabindex="0"
	use:navigator={{
		items,
		fields: internalFields,
		enabled: true,
		indices: cursor
	}}
	onselect={handle}
	onmove={handle}
	onexpand={handle}
	oncollapse={handle}
>
	{#each items as item, index}
		{@const hasItems = item[internalFields.children] && item[internalFields.children].length > 0}
		{@const itemFields = { ...internalFields, ...(internalFields.fields ?? internalFields) }}

		<div
			id={'id-' + index}
			class="flex flex-col"
			class:is-expanded={item[internalFields.isOpen]}
			class:is-selected={item === value}
			data-path={index}
		>
			<Summary fields={internalFields} {using} bind:value={items[index]} />
			{#if hasItems && item[internalFields.isOpen]}
				<list class="flex w-full flex-shrink-0 select-none flex-col" role="listbox" tabindex="-1">
					<ListItems
						bind:items={item[internalFields.children]}
						bind:value
						fields={itemFields}
						{using}
						hierarchy={[index]}
						on:change
					/>
				</list>
				<!-- <List
					bind:items={item[fields.children]}
					bind:value
					fields={itemFields}
					{using}
					on:select
					hierarchy={[index]}
					tabindex="-1"
				/> -->
			{/if}
		</div>
	{/each}
</accordion>
