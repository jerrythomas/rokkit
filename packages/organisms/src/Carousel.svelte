<script>
	import { equals } from 'ramda'
	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 * @property {any} [items]
	 * @property {any} value
	 */

	/** @type {Props} */
	let { class: className = '', items = [], value = $bindable() } = $props()
	let currentIndex = $state(0)

	function handleKey(e, index) {
		const prevIndex = currentIndex
		if (['Enter', ' '].includes(e.key)) {
			currentIndex = index
		} else {
			if (e.key === 'ArrowRight') {
				currentIndex = currentIndex === items.length - 1 ? 0 : currentIndex + 1
			} else if (e.key === 'ArrowLeft') {
				currentIndex = currentIndex === 0 ? items.length - 1 : currentIndex - 1
			}
		}
		if (currentIndex !== prevIndex) {
			e.preventDefault()
			e.stopPropagation()
		}
	}
	$effect(() => {
		value = items[currentIndex]
	})
</script>

{#if items.length === 0}
	<p>No items to display</p>
{:else}
	<carousel class="flex w-full flex-col gap-3 {className}">
		<slide class="flex flex-col">
			<img src={value.src} alt={value.alt} />
			<p>{value.text}</p>
		</slide>
		<dot-nav role="radiogroup">
			{#each items as item, index}
				<dot
					role="radio"
					aria-checked={equals(currentIndex, index)}
					aria-label={`Slide ${index + 1}`}
					onclick={() => (currentIndex = index)}
					onkeydown={(e) => handleKey(e, index)}
					tabindex={0}
				></dot>
			{/each}
		</dot-nav>
	</carousel>
{/if}
