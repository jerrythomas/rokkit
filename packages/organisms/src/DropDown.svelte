<script>
	import { defaultFields, defaultStateIcons, createEmitter } from '@rokkit/core'
	import { dismissable } from '@rokkit/actions'
	import { Icon, Slider } from '@rokkit/atoms'
	import { Item } from '@rokkit/molecules'
	import List from './List.svelte'

	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 * @property {any} [options]
	 * @property {import('@rokkit/core').FieldMapping} [fields]
	 * @property {any} [using]
	 * @property {any} [value]
	 * @property {any} [title]
	 * @property {any} [icon]
	 * @property {boolean} [small]
	 */

	/** @type {Props} */
	let {
		class: className = '',
		options = [],
		fields = $bindable(defaultFields),
		using = $bindable({ default: Item }),
		value = $bindable(null),
		title = null,
		icon = null,
		small = false,
		...events
	} = $props()

	// $effect.pre(() => {
	// using = { default: Item, ...using }
	// fields = { ...defaultFields, ...fields }
	// })
	let internalUsing = $derived({ default: Item, ...using })
	let internalFields = $derived({ ...defaultFields, ...fields })

	let emitter = createEmitter(events, ['change'])
	let offsetTop = $state(0)
	let open = $state(false)
	let icons = defaultStateIcons.selector

	function handleSelect(data) {
		open = false
		emitter.change(data)
	}
</script>

<!-- svelte-ignore a11y_no_noninteractive_tabindex -->
<drop-down
	class="dropdown relative flex cursor-pointer select-none {className}"
	class:open
	aria-haspopup="true"
	aria-controls="menu"
>
	<button
		class="flex items-center"
		bind:clientHeight={offsetTop}
		tabindex="0"
		use:dismissable
		ondismiss={() => (open = false)}
		onfocus={() => (open = true)}
		onblur={() => (open = false)}
	>
		<span class="flex items-center">
			{#if icon !== null}
				<Icon name={icon} />
			{/if}
			{#if !small && title}
				<p class="flex w-full">{title}</p>
			{/if}
			{#if open}
				<Icon name={icons.opened} />
			{:else}
				<Icon name={icons.closed} />
			{/if}
		</span>
	</button>
	{#if open}
		<Slider top={offsetTop + 4}>
			<List
				items={options}
				fields={internalFields}
				using={internalUsing}
				bind:value
				onselect={handleSelect}
				tabindex="-1"
			/>
		</Slider>
	{/if}
</drop-down>
