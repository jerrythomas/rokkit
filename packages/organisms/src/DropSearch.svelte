<script>
	import { defaultFields } from '@rokkit/core'
	import List from './List.svelte'
	import { Slider } from '@rokkit/atoms'
	import { Item } from '@rokkit/molecules'

	export let items
	export let value = null
	/** @type {import('@rokkit/core').FieldMapping} */
	export let fields = defaultFields
	export let using = { default: Item }

	$: using = { default: Item, ...using }
	$: fields = { ...defaultFields, ...fields }

	let opened = false
	let search
	let searchBox

	// on search change filter list
	function handleSelect(event) {
		console.log('selected', event)
		value = event.detail
		search = event.detail[fields.text]
	}
</script>

<!-- svelte-ignore a11y-no-noninteractive-tabindex -->
<div
	class="dropdown relative h-12 w-full flex flex-col outline-none"
	tabindex={0}
>
	<div class="selected-item h-12 flex flex-shrink-0 items-center pl-4">
		<span class="flex flex-grow">
			<input
				type="text"
				class="w-full border-none bg-transparent p-0"
				bind:value={search}
				bind:this={searchBox}
				on:focus={() => {
					opened = true
					searchBox.select()
				}}
				on:blur={() => {
					opened = false
				}}
			/>
		</span>
		{#if opened}
			<icon class="selector-opened" />
		{:else}
			<icon class="selector-closed" />
		{/if}
	</div>

	{#if opened}
		<Slider>
			<List
				bind:items
				bind:value
				{fields}
				{using}
				on:select={handleSelect}
				on:change
			/>
		</Slider>
	{/if}
</div>
