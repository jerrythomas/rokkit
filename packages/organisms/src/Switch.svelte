<script>
	import { equals } from 'ramda'
	import { defaultFields, getComponent, createEmitter } from '@rokkit/core'
	import { Item } from '@rokkit/molecules'
	import { navigator } from '@rokkit/actions'

	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 * @property {any} value
	 * @property {Array<any>} [options]
	 * @property {import('@rokkit/core').FieldMapping} [fields]
	 * @property {any} [using]
	 * @property {boolean} [compact]
	 * @property {boolean} [disabled]
	 */

	/** @type {Props} */
	let {
		class: className = '',
		value = $bindable(),
		options = [false, true],
		fields = $bindable(defaultFields),
		using = $bindable({}),
		compact = false,
		disabled = false,
		...events
	} = $props()

	let cursor = $state([])
	let emitter = createEmitter(events, ['change'])
	function handleNav(event) {
		if (disabled) return
		if (event.detail.node === value) return
		value = event.detail.node
		cursor = event.detail.path
		emitter.change({ item: value, indices: cursor })
	}

	let useComponent = $derived(!options.every((item) => [false, true].includes(item)))
	$effect.pre(() => {
		fields = { ...defaultFields, ...fields }
		using = { default: Item, ...using }
	})
</script>

{#if !Array.isArray(options) || options.length < 2}
	<error>Items should be an array with at least two items.</error>
{:else}
	<toggle-switch
		class="flex items-center {className}"
		class:is-off={options.length === 2 && equals(value, options[0])}
		class:is-on={options.length === 2 && equals(value, options[1])}
		class:compact
		aria-label="Toggle Switch"
		aria-orientation="horizontal"
		aria-disabled={disabled}
		tabindex="0"
		role="listbox"
		use:navigator={{
			items: options,
			fields,
			vertical: false,
			indices: cursor
		}}
		onmove={handleNav}
		onselect={handleNav}
	>
		{#each options as item, index (item)}
			{@const component = useComponent ? getComponent(item, fields, using) : null}
			<item class="relative" role="option" aria-selected={equals(item, value)} data-path={index}>
				{#if equals(item, value)}
					<indicator class="absolute bottom-0 left-0 right-0 top-0"></indicator>
				{/if}
				{#if component}
					{@const SvelteComponent = component}
					<SvelteComponent value={item} {fields} />
				{/if}
			</item>
		{/each}
	</toggle-switch>
{/if}
