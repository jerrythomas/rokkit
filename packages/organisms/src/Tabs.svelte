<script>
	import { createEventDispatcher } from 'svelte'
	import { defaultFields, defaultStateIcons } from '@rokkit/core'
	import { navigator } from '@rokkit/actions'
	import { Icon } from '@rokkit/atoms'
	import { ItemWrapper, Item } from '@rokkit/molecules'

	const dispatch = createEventDispatcher()

	let className = ''
	export { className as class }
	export let options = []
	/** @type {import('@rokkit/core').FieldMapping} */
	export let fields = {}
	export let using
	export let value = null
	export let below = false
	export let align = 'left'
	export let editable = false
	export let icons = defaultStateIcons.action

	let cursor = []

	function handleRemove(item) {
		if (typeof item === Object) {
			item[fields.isDeleted] = true
		} else {
			options = options.filter((i) => i !== item)
		}

		dispatch('remove', { item })
	}
	function handleNav(event) {
		value = event.detail.node
		cursor = event.detail.path

		dispatch('select', { item: value, indices: cursor })
	}
	$: using = { default: Item, ...using }
	$: icons = { ...defaultStateIcons.action, ...icons }
	$: filtered = options.filter((item) => !item[fields.isDeleted])
	$: fields = { ...defaultFields, ...fields }
</script>

<tabs
	class="flex w-full {className}"
	class:is-below={below}
	class:justify-center={align == 'center'}
	class:justify-end={align == 'right'}
	tabindex="0"
	role="listbox"
	use:navigator={{
		items: options,
		fields,
		vertical: false,
		indices: cursor
	}}
	on:move={handleNav}
	on:select={handleNav}
>
	{#each filtered as item, index}
		<ItemWrapper
			value={item}
			{index}
			{fields}
			{using}
			removable={editable}
			selected={item === value}
			{icons}
			class="tab"
			on:remove={handleRemove}
		/>
	{/each}
	{#if editable}
		<Icon
			name="add"
			role="button"
			label="Add Tab"
			size="small"
			on:click={() => dispatch('add')}
		/>
	{/if}
</tabs>
