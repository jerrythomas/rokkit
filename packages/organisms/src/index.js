export * from './lib'
export { default as Accordion } from './Accordion.svelte'
export { default as DropDown } from './DropDown.svelte'
export { default as DropSearch } from './DropSearch.svelte'
export { default as DataEditor } from './DataEditor.svelte'
export { default as Form } from './Form.svelte'
export { default as List } from './List.svelte'
export { default as MultiSelect } from './MultiSelect.svelte'
export { default as NestedList } from './NestedList.svelte'
export { default as Select } from './Select.svelte'
export { default as Switch } from './Switch.svelte'
export { default as Tabs } from './Tabs.svelte'
export { default as Tree } from './Tree.svelte'
export { default as InputField } from './InputField.svelte'
export { default as ListEditor } from './ListEditor.svelte'
export { default as VirtualSelect } from './VirtualSelect.svelte'
export { default as VirtualList } from './VirtualList.svelte'
export { default as NestedEditor } from './NestedEditor.svelte'
export { default as Table } from './Table.svelte'
export { default as TreeTable } from './TreeTable.svelte'
