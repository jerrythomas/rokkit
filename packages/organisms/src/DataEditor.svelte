<script>
	import { run } from 'svelte/legacy'

	import { setContext } from 'svelte'
	import { writable } from 'svelte/store'
	import { componentTypes } from './types'
	import Wrapper from './wrappers/Wrapper.svelte'
	import { Item } from '@rokkit/molecules'
	import Tabs from './Tabs.svelte'
	import FieldLayout from './FieldLayout.svelte'
	import { createEventDispatcher } from 'svelte'

	const dispatch = createEventDispatcher()
	const registry = writable({})
	setContext('registry', registry)

	import { deriveSchemaFromValue, deriveLayoutFromValue, getSchemaWithLayout } from './lib'

	/**
	 * @typedef {Object} Props
	 * @property {any} value
	 * @property {any} [schema]
	 * @property {any} [layout]
	 * @property {any} [using]
	 */

	/** @type {Props} */
	let {
		value = $bindable(),
		schema = $bindable(null),
		layout = $bindable(null),
		using = {}
	} = $props()

	let schemaWithLayout = $derived(getSchemaWithLayout(schema, layout))

	function handle() {
		dispatch('change', value)
	}
	$effect(() => {
		registry.set({
			editors: { ...componentTypes, ...using?.editors },
			components: { default: Item, ...using?.components },
			wrappers: { default: Wrapper, ...using?.wrappers },
			navigators: { default: Tabs, ...using?.navigators }
		})
	})
	$effect(() => {
		if (!schema) schema = deriveSchemaFromValue(value)
	})
	$effect(() => {
		if (!layout) layout = deriveLayoutFromValue(value)
	})
</script>

<FieldLayout schema={schemaWithLayout} bind:value on:change={handle} />
