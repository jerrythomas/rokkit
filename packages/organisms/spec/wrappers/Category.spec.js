import { describe, expect, it, beforeEach } from 'vitest'
import { cleanup, render } from '@testing-library/svelte'
import { tick } from 'svelte'
import { writable } from 'svelte/store'
import { Wrapper, Category } from '../../src/wrappers'
import Switch from '../../src/Switch.svelte'
import { Item } from '@rokkit/molecules'
import Tabs from '../../src/Tabs.svelte'

describe('Category.svelte', () => {
	const items = ['one', 'two', 'three']
	const using = {
		editors: {},
		components: {},
		wrappers: {},
		navigators: {}
	}
	const registry = writable({
		editors: { ...using.editors },
		components: { default: Item, ...using.components },
		wrappers: { default: Wrapper, ...using.wrappers },
		navigators: { default: Tabs, ...using.navigators }
	})

	beforeEach(() => {
		cleanup()
	})

	it('should render using default props', () => {
		const { container } = render(Category, {
			context: new Map([['registry', registry]]),
			props: { options: items }
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
	})

	it('should render using specified type', async () => {
		const { container, component } = render(Category, {
			context: new Map([['registry', registry]]),
			props: { options: items, type: 'horizontal' }
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()

		component.$set({ type: 'section' })
		await tick()
		expect(container).toMatchSnapshot()
	})

	it('should render using specified class', async () => {
		const { container, component } = render(Category, {
			context: new Map([['registry', registry]]),
			props: { options: items, class: 'custom-class' }
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()

		component.$set({ class: 'custom-class-2' })
		await tick()
		expect(container).toMatchSnapshot()
	})
	it('should render with extra props', async () => {
		const { container, component } = render(Category, {
			context: new Map([['registry', registry]]),
			props: { options: items, align: 'center' }
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()

		component.$set({ align: 'right' })
		await tick()
		expect(container).toMatchSnapshot()
	})
	it('should render with alternative navigator', async () => {
		registry.update((state) => {
			state.navigators = { ...state.navigators, Switch }
			return state
		})
		const { container, component } = render(Category, {
			context: new Map([['registry', registry]]),
			props: { options: items, navigator: 'Switch' }
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()

		component.$set({ navigator: 'tabs' })
		await tick()
		expect(container).toMatchSnapshot()
	})
})
