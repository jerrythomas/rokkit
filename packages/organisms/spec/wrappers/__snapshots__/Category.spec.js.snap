// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Category.svelte > should render using default props 1`] = `
<body>
  <div>
    <section
      class=""
    >
      <tabs
        class="flex w-full "
        role="listbox"
        tabindex="0"
      >
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="0"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              one
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="1"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              two
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="2"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              three
            </p>
            
          </item>
           
        </wrap-item>
         
      </tabs>
       
      <field-layout
        class="vertical"
      />
    </section>
    
  </div>
</body>
`;

exports[`Category.svelte > should render using specified class 1`] = `
<body>
  <div>
    <section
      class="custom-class"
    >
      <tabs
        class="flex w-full "
        role="listbox"
        tabindex="0"
      >
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="0"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              one
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="1"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              two
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="2"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              three
            </p>
            
          </item>
           
        </wrap-item>
         
      </tabs>
       
      <field-layout
        class="vertical"
      />
    </section>
    
  </div>
</body>
`;

exports[`Category.svelte > should render using specified class 2`] = `
<body>
  <div>
    <section
      class="custom-class"
    >
      <tabs
        class="flex w-full "
        role="listbox"
        tabindex="0"
      >
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="0"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              one
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="1"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              two
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="2"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              three
            </p>
            
          </item>
           
        </wrap-item>
         
      </tabs>
       
      <field-layout
        class="vertical"
      />
    </section>
    
  </div>
</body>
`;

exports[`Category.svelte > should render using specified type 1`] = `
<body>
  <div>
    <section
      class=""
    >
      <tabs
        class="flex w-full "
        role="listbox"
        tabindex="0"
      >
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="0"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              one
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="1"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              two
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="2"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              three
            </p>
            
          </item>
           
        </wrap-item>
         
      </tabs>
       
      <field-layout
        class="horizontal"
      />
    </section>
    
  </div>
</body>
`;

exports[`Category.svelte > should render using specified type 2`] = `
<body>
  <div>
    <section
      class=""
    >
      <tabs
        class="flex w-full "
        role="listbox"
        tabindex="0"
      >
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="0"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              one
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="1"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              two
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="2"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              three
            </p>
            
          </item>
           
        </wrap-item>
         
      </tabs>
       
      <field-layout
        class="horizontal"
      />
    </section>
    
  </div>
</body>
`;

exports[`Category.svelte > should render with alternative navigator 1`] = `
<body>
  <div>
    <section
      class=""
    >
      <toggle-switch
        aria-disabled="false"
        aria-label="Toggle Switch"
        aria-orientation="horizontal"
        class="flex items-center "
        role="listbox"
        tabindex="0"
      >
        <item
          aria-selected="false"
          class="relative"
          data-path="0"
          role="option"
        >
           
           
          <p>
            one
          </p>
          
          
           
        </item>
        <item
          aria-selected="false"
          class="relative"
          data-path="1"
          role="option"
        >
           
           
          <p>
            two
          </p>
          
          
           
        </item>
        <item
          aria-selected="false"
          class="relative"
          data-path="2"
          role="option"
        >
           
           
          <p>
            three
          </p>
          
          
           
        </item>
      </toggle-switch>
      
       
      <field-layout
        class="vertical"
      />
    </section>
    
  </div>
</body>
`;

exports[`Category.svelte > should render with alternative navigator 2`] = `
<body>
  <div>
    <section
      class=""
    >
      <toggle-switch
        aria-disabled="false"
        aria-label="Toggle Switch"
        aria-orientation="horizontal"
        class="flex items-center "
        role="listbox"
        tabindex="0"
      >
        <item
          aria-selected="false"
          class="relative"
          data-path="0"
          role="option"
        >
           
           
          <p>
            one
          </p>
          
          
           
        </item>
        <item
          aria-selected="false"
          class="relative"
          data-path="1"
          role="option"
        >
           
           
          <p>
            two
          </p>
          
          
           
        </item>
        <item
          aria-selected="false"
          class="relative"
          data-path="2"
          role="option"
        >
           
           
          <p>
            three
          </p>
          
          
           
        </item>
      </toggle-switch>
      
       
      <field-layout
        class="vertical"
      />
    </section>
    
  </div>
</body>
`;

exports[`Category.svelte > should render with extra props 1`] = `
<body>
  <div>
    <section
      class=""
    >
      <tabs
        class="flex w-full justify-center"
        role="listbox"
        tabindex="0"
      >
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="0"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              one
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="1"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              two
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="2"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              three
            </p>
            
          </item>
           
        </wrap-item>
         
      </tabs>
       
      <field-layout
        class="vertical"
      />
    </section>
    
  </div>
</body>
`;

exports[`Category.svelte > should render with extra props 2`] = `
<body>
  <div>
    <section
      class=""
    >
      <tabs
        class="flex w-full justify-center"
        role="listbox"
        tabindex="0"
      >
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="0"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              one
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="1"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              two
            </p>
            
          </item>
           
        </wrap-item>
        <wrap-item
          aria-selected="false"
          class="flex flex-row items-center tab"
          data-path="2"
          role="option"
        >
          <item
            class="flex flex-row items-center"
          >
             
            <p>
              three
            </p>
            
          </item>
           
        </wrap-item>
         
      </tabs>
       
      <field-layout
        class="vertical"
      />
    </section>
    
  </div>
</body>
`;
